### PR TITLE
feat: panda-526 add support for task priority

### PR DESCRIPTION
**Pull request by MirkoPopovicVicert**
## Description

https://canvasmedical.atlassian.net/browse/KOALA-4029

This PR adds `priority` field support to Tasks, Commands, and Effects across the Canvas SDK, along with comprehensive example plugins and tests demonstrating the new functionality.

#### New Priority Field Support
- **Task Priority Enum** (`canvas_sdk/v1/data/task.py`)
  - Added `TaskPriority` enum with values: `ROUTINE`, `URGENT`, `STAT`
  - Added `priority` field to Task data model

- **Command Priority Support**
  - Added `priority` parameter to `TaskCommand` 
  - Added `priority` parameter to `ImagingOrderCommand`
  - Added `priority` parameter to `ReferCommand`

- **Effect Priority Support** (`canvas_sdk/effects/task/task.py`)
  - Added `priority` parameter to `AddTask` effect
  - Added `priority` parameter to `UpdateTask` effect

- **Allowed Imports** (`plugin_runner/allowed-module-imports.json`)
  - Added `canvas_sdk.v1.data.task` and `canvas_sdk.v1.data.team` to allowed imports for plugins

---

#### New Plugin: `refill_priority_setter_plugin`
Demonstrates automatic priority management for refill-related tasks.

**Protocols:**
1. **`CreateRefillTaskProtocol`** - Automatically creates URGENT follow-up tasks when refill commands are committed
   - Listens to `REFILL_COMMAND__POST_COMMIT` events
   - Creates tasks assigned to teams with `PROCESS_REFILL_REQUESTS` responsibility
   - Prevents duplicate task creation

2. **`RefillTaskPriorityProtocol`** - Automatically updates task priority based on team responsibilities
   - Listens to `TASK_CREATED`, `TASK_UPDATED`, and `TASK_COMMAND__POST_UPDATE` events
   - Sets priority to URGENT for teams with refill processing responsibilities
   - Handles both task events and command events

#### New Plugin: `example_commands`
Demonstrates priority field usage in command auto-population.

**Protocols:**
1. **`AutoPopulateImagingOrderCommand`** - Auto-populates imaging orders with random priority (ROUTINE, URGENT, STAT, or None)
   - Listens to `IMAGING_ORDER_COMMAND__POST_ORIGINATE` events

2. **`AutoPopulateReferCommand`** - Auto-populates referral commands with random priority
   - Listens to `REFER_COMMAND__POST_ORIGINATE` events

---

#### Test Infrastructure Improvements
- **Created `__init__.py` files** for plugins to make them proper Python packages

#### New Test Suites

**`tests/example_commands/`** (11 tests total, split into 3 files):
- `test_imaging_order_command.py` - 5 tests for imaging order protocol
- `test_refer_command.py` - 6 tests for refer command protocol  
- `test_example_commands.py` - Legacy combined tests with factory examples

**`tests/refill_priority_setter_plugin/`** (17 tests total):
- `test_create_refill_task.py` - 6 tests covering:
  - Task creation for new refills
  - Duplicate prevention
  - Team assignment handling
  - Medication extraction

- `test_priority_setter.py` - 11 tests covering:
  - Multiple event type handling (TASK_CREATED, TASK_UPDATED, TASK_COMMAND__POST_UPDATE)
  - URGENT priority setting for refill teams
  - Edge cases (closed tasks, unassigned tasks, non-refill teams)
  - Helper method validation

#### Test Quality Features
- Proper mocking to isolate protocol logic
- Preserves real enum values (TaskPriority, TeamResponsibility, Priority, ClinicalQuestion) instead of MagicMock objects
- Follows established patterns from existing example plugin tests
- Comprehensive edge case coverage

### CODE REVIEW

**Comment by MirkoPopovicVicert** [comment]
Test Plan: Refill Priority and Urgent Banner Plugins
This outlines the testing steps to verify the interaction between the refill_priority_setter_plugin and the urgent_task_banner_plugin.

Part 1: Prerequisites & Setup
Before starting, ensure the following are configured in your Canvas environment:
1. Plugins Installed:
* Install and enable refill_priority_setter_plugin.
* Install and enable urgent_task_banner_plugin.
2. Create Test Team:
* Navigate to Settings > Teams.
* Ensure a team exists that has the "Process refill requests from a pharmacy" responsibility.
* (Note: Using a team like "All Responsibilities" is perfectly acceptable for this test, as it fulfills the prerequisite.)
3. Select a Test Patient:
* Choose any patient for testing.

Part 2: Test Cases
Scenario 1: Creation of Urgent Task
Goal: Verify that creating a qualifying refill task correctly sets its priority to "Urgent" AND triggers the appearance of the patient banner and metadata update.
1. Navigate to your test patient's chart.
2. Create a new Task.
3. Set the following details:
* Title: Needs prescription refill
* Assign to Team: Select the "All Responsibilities" team (or your configured "Refill Team").
4. Save the task.
5. Verification:
* Check Task Priority: Open the task you just created.
* Expected Result: The Priority field is set to Urgent.
* Check Patient Banner: Return to the main patient chart view.
* Expected Result: A banner with the text "Patient has urgent task" is visible.
* Check Patient Metadata: Navigate to the Patient object in the Django admin panel and inspect its metadata.
* Expected Result: The metadata contains a key Urgent refill with the value Yes.
Scenario 2: Removal of Priority and Banner
Goal: Verify that removing the team assignment from the urgent task clears its priority AND triggers the removal of the patient banner and updates the metadata.
1. Use the urgent task created in Scenario 1.
2. Edit the task.
3. Change the Assign to Team field to "Unassigned" (clear the selection).
4. Save the task.
5. Verification:
* Check Task Priority: Open the task again.
* Expected Result: The Priority field is now empty/cleared.
* Check Patient Banner: Return to the patient's chart.
* Expected Result: The "Patient has urgent task" banner is now gone.
* Check Patient Metadata: Navigate to the Patient object in the Django admin panel and inspect its metadata.
* Expected Result: The metadata key Urgent refill now has the value No.

https://github.com/user-attachments/assets/48b28de6-70ff-4324-a380-f0ac61852ab1

**Review by filips-vicert — COMMENTED** [review]
LGTM

**Review by nmpsilva — COMMENTED** [review]
I think this looks good, although I have a few questions and think  you could simplify it quite a bit by being a bit less defensive. I.e., moving away from `getattr` and stop catching `Exception` which is generally seen as too broad a clause to actually be helpful.

File: test_plugins/refill_priority_setter_plugin/protocols/priority_setter.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        try:
+            task = Task.objects.get(id=task_id)
+            log.info(
+                f"Fetched task: {json.dumps(task.__dict__, indent=2, default=json_default_serializer)}"
+            )
+            return task
+        except getattr(Task, "DoesNotExist", Exception):
+            log.error(f"Task with id={task_id} does not exist.")
+        except Exception as e:
+            log.exception(f"Unexpected error fetching task id={task_id}: {e}")
```
[the remark this answers is not in the document]
Comment (MirkoPopovicVicert) [reply test_plugins/refill_priority_setter_plugin/protocols/priority_setter.py:112]: I didn't know what exceptions to expect, so I covered all of them for debugging purposes. I wanted to make sure any error was logged rather than potentially crashing the plugin silently.

**Review by ruipbatista — COMMENTED** [review]
Looks good, but the example plugin is not following the new plugin structure and is missing tests.

**Comment by nmpsilva** [comment]
Everything looks good, I'd just update the example plugins to follow the updated pattern and add some tests for them.

**Comment by rmagier1** [comment]
> Everything looks good, I'd just update the example plugins to follow the updated pattern and add some tests for them.

@nmpsilva Okay updated this to have tests. I also saw the pattern that Mary pushed was add all the tests for all plugins in the test subfolder in example-plugins, so I couldn't just use canvas init. But it works

**Review by nmpsilva — APPROVED** [review]
LG!

I'd rather us having tests from inside the plugin package itself, but we're still working on that.

**Review by jamagalhaes — APPROVED** [review]

**Review by beaugunderson — APPROVED** [review]
